### PR TITLE
[Tiny PR] Paste: do not force blocks for empty rich text

### LINKED ISSUE #16990: Pasting unformatted text removes formatting and converts heading to paragraph
**Describe the bug**

If I paste an unformatted word (eg. copied from the URL bar) into an empty heading block or a paragraph with styling applied, all styling is reset and the block is converted to a paragraph.

**To reproduce**

1. Create a _Heading_ block
2. Copy an unformatted word, eg a word from the URL bar of the browser
3. Paste it into the block
4. Notice that the block is now a paragraph.

Same happens if you paste it into the styled placeholder paragraph block in _Media & Text_.

**Expected behavior**

If I paste an unformatted word I would expect it to keep the previous styling and block type rather than resetting everything to an unformatted paragraph.

**Desktop (please complete the following information):**
 - OS: macOS
 - Browser Chrome
 - Version 76

**Additional context**

This works as expected with [v4.7.1](https://testgutenberg.com/) and v5.5.0. It does not work in v5.9.1, v6.2.0 and master branch.

### PR DESCRIPTION
**Pull request by ellatrix**
## Description

Fixes #16990. There's probably similar older issues.
Also blocks introducing caption splitting as you wouldn't be able to paste in an empty caption.

Reverts some of #3326, introduced 2 years ago for embedding on URL paste. I've solved that problem differently here.

## How has this been tested?

See #16990. Create an empty heading. Copy paste a piece of the URL in the address bar. Expect a heading with the pasted text. On master it would be converted to a paragraph.
Paste an embeddable URL in an empty paragraph. It should be converted to an embed block.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate.

### CODE REVIEW

**Comment by oxyc** [comment]
I can confirm (with limited testing) that this fixes my issue.

**Review by peterwilsoncc — COMMENTED** [review]
✅ Testing this branch using the same techniques I used in https://github.com/WordPress/gutenberg/issues/16990#issuecomment-523747832 works as expected.

I don't know enough about `__unstableEmbedURLOnPaste` or the rich text component to provide a code review I'm afraid. If I find myself at a loose end, I'll dive in to the code to see if I can figure it out but I can't promise that will happen.

File: packages/block-library/src/paragraph/edit.js [hunk 1]
```diff
@@ -195,6 +195,7 @@ class ParagraphBlock extends Component {
 					onRemove={ onReplace ? () => onReplace( [] ) : undefined }
 					aria-label={ content ? __( 'Paragraph block' ) : __( 'Empty block; start writing or type forward slash to choose a block' ) }
 					placeholder={ placeholder || __( 'Start writing or type / to choose a block' ) }
+					__unstableEmbedURLOnPaste
```
Comment (desaiuditd) [inline packages/block-library/src/paragraph/edit.js]: Is this prop newly introduced in this PR? (Could not find its reference anywhere else in the codebase.) What is its purpose?

Not sure, what's the convention for such unstable new props. But can we add some doc on this? Maybe inline comment, readme, somewhere else? Maybe better prop naming to reflect its use case better?

**Review by peterwilsoncc — APPROVED** [review]
I've done some further research on paste handling and modes since my earlier comment. Now feel confident enough to approve this.

Further testing based on my enhanced knowledge worked.

File: packages/block-library/src/paragraph/edit.js (hunk 1 above)
Comment (ellatrix) [reply packages/block-library/src/paragraph/edit.js]: I can provide an inline comment, but I thought the name is quite clear: it enables embedding when the user pastes a URL. This prop should normally only be used on the core paragraph block. We shouldn't document unstable props in a readme file.

> An unstable API is one which serves as a means to an end. It is not desired to ever be converted into a public API.

https://developer.wordpress.org/block-editor/contributors/develop/coding-guidelines/

**Comment by ellatrix** [comment]
@peterwilsoncc Thank you very much! I'm happy you felt like exploring raw handling. :)

**Pull request merged**